olp-cpp-sdk-dataservice-read/src/ReleaseDependencyResolver.h

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (C) 2020 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#pragma once
+
+#include <map>
+#include <memory>
+#include <string>
+
+#include <olp/core/cache/KeyValueCache.h>
+#include <olp/core/geo/Types.h>
+#include <olp/core/geo/tiling/TileKey.h>
+#include "repositories/DataCacheRepository.h"
+#include "repositories/DataRepository.h"
+#include "repositories/PartitionsCacheRepository.h"
+#include "repositories/PartitionsRepository.h"
+
+namespace olp {
+namespace dataservice {
+namespace read {
+
+class ReleaseDependencyResolver {
+ public:
+  // group quad trees and protected tiles. Check if for each quad tree we want
+  // to release all protected keys associated, if true add this quad tree to
+  // released keys
+  static cache::KeyValueCache::KeyListType GenerateKeysToRelease(
```

Comment:
This class setup is very strange. A static method that creates a temporary instance of itself to generate some data and copies internally not very cheap data. If it is only temporary class then do not copy but use references and remove the static from this method.

Reply:
Static method removed